cmd/warpforge/main_test.go

```diff
@@ -28,6 +28,25 @@ func getTestWorkspaceStack(t *testing.T) []*workspace.Workspace {
 	return wss
 }
 
+type tagset map[string]struct{}
```

Comment:
I like this whole bit :+1:   A good terse little abstraction for this feels nice and makes it clearer what's going on.

Reply:
Same, though a nit... we might want to use this elsewhere as well. Maybe worth extracting the tags bits into a util package?

Could happen later though.

Reply:
Vote for later.  I appreciate that this is a very small hunk of code, and that it doesn't have to take itself too seriously because it has exactly one user that's very local